### PR TITLE
Added documentation for Rubi

### PR DESCRIPTION
**Pull request by czgdp1807**
#### References to other Issues or PRs

https://github.com/sympy/sympy/pull/13694
Fixes #13261 

#### Release Notes

NO ENTRY

### CODE REVIEW

File: doc/src/modules/integrals/rubi/about.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+important role in making RUBI more robust and fast. As it works on the given
+set of rules. MatchPy is a pattern matching library in Python which has matching
+capabilities similar to Mathematica. But because MatchPy is only implemented in
+Python3.6 so RUBI is not supported by Python versions less than 3.6.
+
+Note
+----
+
+Please note that this module is experimental. Therefor, API may change unexpectedly
+between SymPy versions.
```
Comment (czgdp1807) [inline doc/src/modules/integrals/rubi/about.rst]: I have added the note here. @Upabjojr

Comment (czgdp1807) [reply doc/src/modules/integrals/rubi/about.rst]: ping @Upabjojr

Comment (czgdp1807) [reply doc/src/modules/integrals/rubi/about.rst]: ping @Upabjojr

**Comment by czgdp1807** [comment]
@jksuom Can you please review this PR?

File: doc/src/modules/integrals/rubi/about.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
+About Rule Based Integrator
+===========================
+
+About it
```
Comment (jksuom) [inline doc/src/modules/integrals/rubi/about.rst]: Could we remove this? It only repeats the heading above.

File: doc/src/modules/integrals/rubi/about.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+--------
+
+A RUle Based Integrator nicknamed RUBI is a module in sympy.integrals. It is an
```
Comment (Upabjojr) [inline doc/src/modules/integrals/rubi/about.rst]: RUBI is an external project, we are importing the rules. It sounds a bit weird if you start like this a sentence.

File: doc/src/modules/integrals/rubi/about.rst [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+implementation of more than 6000 rules to determine antiderivative of a wide
+variety of mathematical expressions. RUBI utilizes a set of well defined rules
+which makes it smart to present the results in a more symmetric and simplified
```
Comment (Upabjojr) [inline doc/src/modules/integrals/rubi/about.rst]: sentence is too long compared to its contents

File: doc/src/modules/integrals/rubi/about.rst (hunk 1 above)
Comment (Upabjojr) [reply doc/src/modules/integrals/rubi/about.rst]: OK this looks good.

File: doc/src/modules/integrals/rubi/pattern_matcher.rst [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+Matchpy
+-------
+
+**MatchPy** implements pattern matching in python. It is similar to the 
+implementation in Mathematica. A detailed example of how you can use matchpy 
+can be found in its documentation (see the link below). In addition to the 
+basic matching algorithm, there are data structures that can be used for more
+efficient many-to-one matching like the ManyToOneMatcher and the 
+DiscriminationNet.
```
Comment (Upabjojr) [inline doc/src/modules/integrals/rubi/pattern_matcher.rst]: We are only using `ManyToOneMatcher`... strange sentence.

File: doc/src/modules/integrals/rubi/utility_functions.rst [hunk 6]
```diff
@@ -0,0 +1,142 @@
+Utility Functions
+=================
+
+Utility functions or helper functions play an essential role in RUBI apart 
+from pattern matching. It is required to facilitate integration methods to 
+check conditions or use other integration techniques.
```
Comment (Upabjojr) [inline doc/src/modules/integrals/rubi/utility_functions.rst]: Poor explanation.

**Comment by Upabjojr** [comment]
Maybe the sentences could be better written. There are long sentences with little content.

**Pull request closed without merging**